constellation-app/constellation
# Remember the last saved or last opened file location

### Description of the Change
The last directory accessed for opening or saving a graph is now saved and remembered between sessions as well as within a session. Changed the preference to `Remember Open/Save Location` in `Setup`->`Options`->`CONSTELLATION`->`Save Location`, which is enabled by default. When this is not set, both the open and save locations will be the user's home directory.


### Alternate Designs
N/A


### Why Should This Be In Core?
Better UX


### Benefits
Saves time.


### Possible Drawbacks
N/A


### Verification Process

1. First check the preference to remember the last saved location is turned on. Go to Options > Constellation > `Open/Save Location`. Ensure `Remember Open/Save location` is ticked
2. Create and save a new graph. Take note of the saved location.
3. Open up another new graph, draw a node and click to save (Using Save, Save as or Ctrl+s). Verify the previous location is opened.
4. Try to open a previously saved STAR file using File-> open on or Ctrl+o, and verify the same location is opened on the Open File window.
5. Try to save a file now, and verify the previously opened location is opened up in the Save file window.
6. Close constellation, noting the last opened or saved location. Run it again and repeat again. Verify both Open and Save windows open up with this location.
7. Open a file in the Welcome page and repeat the same above.
8. Repeat again, changing the opened and saved location in between.
9. Repeat again with the preference unticked and verify the user's home directory is opened up.


### Applicable Issues
https://github.com/constellation-app/constellation/issues/1098
